panel/io/jupyter_server_extension.py

```diff
@@ -1,21 +1,140 @@
 from urllib.parse import urljoin
 
-from tornado.web import StaticFileHandler
+import tornado
 
-from .resources import DIST_DIR
+from bokeh.command.util import build_single_handler_application
+from bokeh.embed.bundle import extension_dirs
+from bokeh.embed.server import server_html_page_for_session
+from bokeh.protocol import Protocol
+from bokeh.protocol.exceptions import MessageError, ProtocolError, ValidationError
+from bokeh.protocol.receiver import Receiver
+from bokeh.server.connection import ServerConnection
+from bokeh.server.contexts import ApplicationContext
+from bokeh.server.protocol_handler import ProtocolHandler
+from bokeh.server.views.doc_handler import DocHandler
+from bokeh.server.views.multi_root_static_handler import MultiRootStaticHandler
+from bokeh.server.views.static_handler import StaticHandler
+from bokeh.server.views.ws import WSHandler
+from bokeh.server.auth_provider import NullAuth
+from tornado.web import StaticFileHandler, RequestHandler
 
+from ..util import edit_readonly
+from .state import state
+from .resources import DIST_DIR, Resources
+
+
+_sessions = {
+}
+
+_RESOURCES = None
+
+
+class PanelHandler(DocHandler):
+
+    def __init__(self, *args, **kw):
+        kw['application_context'] = None
+        kw['bokeh_websocket_path'] = None
+        super().__init__(*args, **kw)
+        app = self.application
+        app.auth_provider = NullAuth()
+        app.generate_session_ids = True
+        app.sign_sessions = False
+        app.include_headers = None
+        app.include_cookies = None
+        app.exclude_headers = None
+        app.exclude_cookies = None
+        app.session_token_expiration = 300
+        app.secret_key = None
```

Comment:
Should find a way to clean this up, basically we have to mock a BokehTornado application since we are running these handlers on a Jupyter server.